packages/terra-date-picker/src/DatePicker.jsx

```diff
@@ -137,15 +138,18 @@ class DatePicker extends React.Component {
   constructor(props) {
     super(props);
 
+    const activeBreakpointOnMount = activeBreakpointForSize(window.innerWidth);
```

Comment:
What happens if you remove this initial set?

I'm not sure if the constructor is a safe place to analyze the window. The component has not technically mounted yet. The responsive element should immediately re-render the component when mounting. If this default is removed does the component flash?

Reply:
I added this to prevent the need for a re-render to when it's being mounted and the responsive-element returns the same active breakpoint.

Reply:
I think the odds are good that this would prevent an extra render most of the time, but it is a gamble. Assigning a breakpoint based on the window width in the constructor prior to the component mounting has no guarantee it will mount with the correct breakpoint.

Reply:
> Assigning a breakpoint based on the window width in the constructor prior to the component mounting has no guarantee it will mount with the correct breakpoint.

I'm not sure I understand when the window's width would be different than the width it has on mount. Could you provide some examples or context as to the instances you are thinking of?

My thought it this would be toggling against a true or false boolean so if the breakpoint isn't tiny or small like defined in the window's width when it's mounted, the responsive element would send the correct breakpoint size back.

Reply:
Variables initialized in the constructor are created when the component is declared not when the component is mounted. 

Here a demo that will demonstrate this. Load this onto a page and then resize the browser. You'll see that each component was initialized with the the breakpoint of the window, but when they mount they'll still have the constructor breakpoint and not the actual breakpoint.

```jsx
/* eslint-disable max-classes-per-file */
import React, { useContext } from 'react';
import { ActiveBreakpointContext, ActiveBreakpointProvider, activeBreakpointForSize } from 'terra-breakpoints';

class Example extends React.Component {
  constructor() {
    super();

    this.constructorBreakpoint = activeBreakpointForSize(window.innerWidth);
  }

  render() {
    return (
      <div>
        {`Constructor Breakpoint: ${this.constructorBreakpoint}`}
      </div>
    );
  }
}

const BreakpointsExample = (props) => {
  const breakpoint = useContext(ActiveBreakpointContext);

  return (
    <div>
      {`Actual Breakpoint: ${breakpoint}`}
      {props[breakpoint]}
    </div>
  );
};

const Provider = () => {
  const tiny = <Example />;
  const small = <Example />;
  const medium = <Example />;
  const large = <Example />;
  const huge = <Example />;
  const enormous = <Example />;

  return (
    <ActiveBreakpointProvider>
      <BreakpointsExample tiny={tiny} small={small} medium={medium} large={large} huge={huge} enormous={enormous} />
    </ActiveBreakpointProvider>
  );
};

export default Provider;
```

Reply:
I'm looking through the example you provided, and I see where this could be a concern with a component that is store off it's width when it's initialized, but I added a this comparisons which has similar implementation to the changes I made in this PR and the breakpoints are always matching the `BreakpointsExample` component you linked out.

```js
class Example2 extends React.Component {
  constructor() {
    super();
    this.state = { breakpoint: activeBreakpointForSize(window.innerWidth) };
    this.handleBreakpointChange = this.handleBreakpointChange.bind(this);
  }

  handleBreakpointChange(updatedBreakpoint) {
    this.setState({ breakpoint: updatedBreakpoint });
  }

  render() {
    return (
      <ResponsiveElement responsiveTo="window" onChange={this.handleBreakpointChange}>
        <div>
          {`My Example of Breakpoint in State: ${this.state.breakpoint}`}
        </div>
      </ResponsiveElement>
    );
  }
}
```

I think your concerns should be addressed with the onChange callback passed to the Responsive Element: https://github.com/cerner/terra-framework/pull/1042/files#diff-06f07d3cca7012107347dc63be00c147R409

Reply:
I'm hesitant to include performance optimizations that are based on random chance. In this case the end result should be the same with or without the dice roll in the constructor because the responsive element will invoke a re-render before the dom paints. I do think this has a good chance of being correct most of the time, but it's important to understand that it is not guaranteed.